be/src/storage/lake/lake_primary_index.cpp

```diff
@@ -111,7 +111,9 @@ Status LakePrimaryIndex::_do_lake_load(TabletManager* tablet_mgr, const TabletMe
         case PersistentIndexTypePB::CLOUD_NATIVE: {
             _persistent_index = std::make_unique<LakePersistentIndex>(tablet_mgr, metadata->id());
             set_enable_persistent_index(true);
-            return Status::OK();
+            auto* lake_persistent_index = dynamic_cast<LakePersistentIndex*>(_persistent_index.get());
+            RETURN_IF_ERROR(lake_persistent_index->init(metadata->sstable_meta()));
+            return lake_persistent_index->load_from_lake_tablet(tablet_mgr, metadata, base_version, builder);
```

Comment:
can this init and load_from_lake_tablet be sured to be performed on scan io thread, since it involves a lot of IO work and possibly the remote IO.

Reply:
If the directory of local persistent index is dropped or the tablet is scheduled to another BE, it also have the same logic. What's more, local persistent index needs to recover from version 1 while cloud native persistent index just needs to recover from max_sstable_version.

Reply:
my concern is which type of threads will this code be executed. Do we have any control of that.

Reply:
`Rowset::load_segments` is used to fetch the footer of segments. And after the returned `SegmentIterator` is used to fetch the chunk data. It is single thread.